GPy/inference/latent_function_inference/expectation_propagation.py

```diff
@@ -244,48 +256,46 @@ def expectation_propagation(self, K, Y, likelihood, Y_metadata):
         marg_moments = marginalMoments(num_data)
         cav_params = cavityParams(num_data)
-        ga_approx, post_params = self._init_approximations(K, num_data)
+        ga_approx, post_params = self._init_approximations(mean_prior, K, num_data)
 
         #Approximation
         stop = False
         iterations = 0
         while not stop and (iterations < self.max_iters):
-            self._local_updates(num_data, cav_params, post_params, marg_moments, ga_approx, likelihood, Y, Y_metadata)
+            self._local_updates(mean_prior, K, num_data, cav_params, post_params, marg_moments, ga_approx, likelihood, Y, Y_metadata)
 
             #(re) compute Sigma and mu using full Cholesky decompy
-            post_params = posteriorParams._recompute(K, ga_approx)
+            post_params = posteriorParams._recompute(mean_prior, K, ga_approx)
 
             #monitor convergence
             if iterations > 0:
                 stop = self._stop_criteria(ga_approx)
             self.ga_approx_old = gaussianApproximation(ga_approx.v.copy(), ga_approx.tau.copy())
             iterations += 1
 
-        # Z_tilde after removing the terms that can lead to infinite terms due to tau_tilde close to zero.
-        # This terms cancel with the coreresponding terms in the marginal loglikelihood
         log_Z_tilde = self._log_Z_tilde(marg_moments, ga_approx, cav_params)
-                         # - 0.5*np.log(tau_tilde) + 0.5*(v_tilde*v_tilde*1./tau_tilde)
+
         return (post_params, ga_approx, cav_params, log_Z_tilde)
 
-    def _init_approximations(self, K, num_data):
+    def _init_approximations(self, mean_prior, K, num_data):
         #initial values - Gaussian factors
         #Initial values - Posterior distribution parameters: q(f|X,Y) = N(f|mu,Sigma)
         if self.ga_approx_old is None:
             v_tilde, tau_tilde = np.zeros((2, num_data))
             ga_approx = gaussianApproximation(v_tilde, tau_tilde)
             Sigma = K.copy()
             diag.add(Sigma, 1e-7)
-            mu = np.zeros(num_data)
+            mu = mean_prior
             post_params = posteriorParams(mu, Sigma)
         else:
             assert self.ga_approx_old.v.size == num_data, "data size mis-match: did you change the data? try resetting!"
             ga_approx = gaussianApproximation(self.ga_approx_old.v, self.ga_approx_old.tau)
-            post_params = posteriorParams._recompute(K, ga_approx)
+            post_params = posteriorParams._recompute(mean_prior, K, ga_approx)
             diag.add(post_params.Sigma, 1e-7)
             # TODO: Check the log-marginal under both conditions and choose the best one
         return (ga_approx, post_params)
 
-    def _local_updates(self, num_data, cav_params, post_params, marg_moments, ga_approx, likelihood, Y, Y_metadata, update_order=None):
+    def _local_updates(self, mean_prior, K, num_data, cav_params, post_params, marg_moments, ga_approx, likelihood, Y, Y_metadata, update_order=None):
```

Comment:
Given the other comment, it's not clear to me how mean_prior is being used when doing local updates? Does it belong in here as well?

Reply:
It is not, and also K is not used. I was using them for debugging and somehow forgot to delete them. I will create a new pull request after amending this. Thanks a lot!